backend/src/question/question.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from "@nestjs/common";
+import { Question } from "@prisma/client";
+
+import { PrismaService } from "src/core/prisma.service";
+
+import { DifficultyLevel } from "../../../shared/src/types/base";
+
+@Injectable()
+export class QuestionService {
+  constructor(private prisma: PrismaService) {}
+
+  async getQuestionById(id: number): Promise<Question | null> {
+    return this.prisma.question.findFirst({
```

Comment:
We can use `findUnique` here instead.
```suggestion
    return this.prisma.question.findUnique({
```